lib/active_force/composite/traversable.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+require 'active_force/association/belongs_to_association'
+require 'active_force/association/has_many_association'
+require 'active_force/association/has_one_association'
+require 'set'
+
+module ActiveForce
+  module Composite
+    #
+    # Include on SObject. Exposes methods for convenient traversing
+    # that return loaded association records that may have been updated or newly built.
+    #
+    module Traversable
+      Relationship = Struct.new(:association) do
```

Comment:
Not 100% happy with this Class and its placement here--it seems ad-hoc (which it kind of is to support the back-assignment of inverse associations).  Although, it might be fine for now until a stronger pattern emerges with how this concern gets used in other Composite applications.